### PR TITLE
Summary tests

### PR DESCRIPTION
**Pull request by seth127**
Adding some basic tests of `bbi nonmem summary` using golden files

### CODE REVIEW

**Comment by seth127** [comment]
One question here is: when I run these locally I have to set the `ROOT_EXECUTION_DIR` environment var like so
```
ROOT_EXECUTION_DIR=. go test -v -run TestSummary
```
because I'm _not_ using the `Scenarios` construct for two reasons:
1. I don't need it because I'm not writing anything to disk
2. It breaks my ability to have the `-update_summary` flag for updating the golden files.

@shairozan we should talk on this and make sure I'm cool in drone, or if maybe there's another way to accomplish the update thing that's a bit slicker.

File: bbi_summary_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	for _, tc := range(testConfigs) {
+
+		commandAndArgs := []string{
+			"nonmem",
+			"summary",
+			filepath.Join(SUMMARY_TEST_DIR, mod, mod+".lst"),
+		}
+
+		if (tc.bbiFlag != "") {
```
Comment (shairozan) [inline bbi_summary_test.go:38]: As a note, go does not require the parentheses

```go
 if tc.bbiFlag != "" {
}
```

is valid and more idiomatic

File: bbi_summary_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		require.Nil(t,err)
+		require.NotEmpty(t,output)
+
+		gtd := GoldenFileTestingDetails{
+			t:               t,
+			outputString:    output,
+			goldenFilePath:  filepath.Join(SUMMARY_TEST_DIR, SUMMARY_GOLD_DIR, mod+".golden"+tc.goldenExt),
+		}
+
+		if *update_summary {
```
Comment (shairozan) [inline bbi_summary_test.go:53]: I do very much like the fact you built in a way to update the golden file without having to manually copy files around

**Review by shairozan — CHANGES_REQUESTED** [review]
We'll want a few more tests than just the happy path for summary. What happens if:

1. The model / directory targeted doesn't exist? 
2. The model / directory exists but the files necessary to run summary did not (IE failed grid submission)

What we want to do is make sure that we _understand_ babylon's behavior. If it's going to panic in these scenarios, let's update Babylon to accommodate these without panics and verify that behavior in the tests.

Overall good reconciliation of the complicated test structures we have for babylon and navigating that to create useful tests for Babylon

**Pull request merged**